src/GafferDispatch/Dispatcher.cpp

```diff
@@ -55,6 +55,51 @@ using namespace IECore;
 using namespace Gaffer;
 using namespace GafferDispatch;
 
+//////////////////////////////////////////////////////////////////////////
+// Internal utilities
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+/// \todo Consider moving this to PlugAlgo and using it in
+/// `Shader::NetworkBuilder::effectiveParameter()`.
+tuple<const Plug *, ConstContextPtr> computedSource( const Plug *plug )
+{
+	plug = plug->source();
+
+	if( auto sw = runTimeCast<const Switch>( plug->node() ) )
+	{
+		if(
+			sw->outPlug() &&
+			( plug == sw->outPlug() || sw->outPlug()->isAncestorOf( plug ) )
+		)
+		{
+			if( auto activeInPlug = sw->activeInPlug( plug ) )
+			{
+				return computedSource( activeInPlug );
+			}
+		}
+	}
+	else if( auto contextProcessor = runTimeCast<const ContextProcessor>( plug->node() ) )
+	{
+		if( plug == contextProcessor->outPlug() )
```

Comment:
Sorry, potentially a silly Q - but  do we need the `isAncestorOf` check here too? Both nodes seemed to create their outputs int he same way.

Reply:
The `isAncestorOf()` check is necessary for `Switch` because `NameSwitch` derived from it, and nests the output `TaskPlug` inside a `NameValuePlug`. That doesn't happen in the `ContextProcessor` case. But we will need the ancestor checks should we promote `computedSource()` to the public API, so I've added that in db8e73e.

Reply:
Thanks @johnhaddon, did you want to squash that commit or merge as is?